### DISCUSSION TITLE
[New] Proposition d'ajout : « Assurer la compatibilité avec les plus anciens appareils et logiciels du parc »

### DISCUSSION THREAD

**Discussion by rlemaire** [cnumr/best-practices#50] [Bonnes pratiques] [answered]
A part peut-être mobile first, il n'y a pas de pratique qui dit que la solution doit être compatible avant d'être légère et rapide. On parle beaucoup d'optimisation mais si le site ne s'affiche correctement pas tout court, on a également une pression à l'obsolescence.

La règle doit être de concevoir le programme pour être compatible avec les configurations les plus contraignantes du parc cible (visible à travers les analytics ou une étude des utilisateurs) qui ont un pourcentage non négligeable de l'audience.

Je pense aux versions successives d'IE pour la décennie 201x, mais rester générique permet de cibler toute plateforme contraignante de ce type dans le futur.

**Comment by tbroyer** [comment]
Attention aux analytics, ça peut être trompeur (https://fr.wikipedia.org/wiki/Biais_des_survivants).

Le _progressive enhancement_ / _graceful degradation_ devrait être encouragé/privilégié, avec potentiellement une approche [_cutting the mustard_](https://web.archive.org/web/20200408052913/https://responsivenews.co.uk/post/18948466399/cutting-the-mustard) pour les navigateurs les plus anciens (ou quand JavaScript est désactivé ou a "planté")

**Comment by DocRoms** [comment]
On pourrait également citer que d'un point de vue qualité, la norme ISO 25010 spécifie de checker ça avec des tests de portabilité ( ou plus précisément d'adaptabilité ) ;) 

Cf https://latavernedutesteur.fr/2018/12/03/types-de-tests-iso-25-010-les-tests-de-portabilite-8-8/

**Comment by ACTLEM** [comment]
Note équipe projet: Ok pour principe, travailler la règle de validation.

**Comment by Christophe-Cl** [comment]
Suggestion : inclure le périmètre des OS à la règle (en sus des versions d'appareils et de logiciels)

**Reply by rlemaire** [reply]
Ca fait partie de logiciel pour moi. Je le préciserai dans la description.

**Comment by ACTLEM** [comment, marked as answer]
Note équipe projet: Ok pour création, attention à la règle de validation

Issue créée: https://github.com/cnumr/best-practices/issues/173
